src/shogun/lib/SGMatrix.h

```diff
@@ -306,9 +309,44 @@ template<class T> class SGMatrix : public SGReferencedData
 		/** Set matrix to a constant */
 		void set_const(T const_elem);
 
-		/** fill matrix with zeros */
+		/** Fill matrix with zeros */
 		void zero();
 
+		/**
+		 * Fill matrix with uniformly distributed randoms
+		 * Float Types: [0, 1)
+		 * Int Types: std::numeric_limit<T>::(min->max)
+		**/
+		template <typename U = T> requires Floating<U> && RngVectorizable<U>
+		void random() {
+			auto r = std::make_unique<CRandom>();
```

Comment:
Let's do it at once with the approach to set the seed. I believe it is crucial or at least quite helpful for unit testing.

Reply:
Of course here and in the other functions as well.

Reply:
Not sure why it picked up the older commit without the `sg_rand` addition when I merged :/ . Will add that in.